front/src/applications/editor/components/InfraErrors/InfraErrorCorrectorModal.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FC, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+
+import { ConfirmModal, useModal } from 'common/BootstrapSNCF/ModalSNCF';
+import { Operation, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import Loader from 'common/Loader';
+import { saveOperations } from 'reducers/editor';
+import { getInfraID } from '../../../../reducers/osrdconf/selectors';
+import { OPERATION_TYPE } from 'applications/editor/tools/types';
+
+const InfraErrorCorrectorModal: FC = () => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const { closeModal } = useModal();
+
+  const infraId = useSelector(getInfraID);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isSaving, setIsSaving] = useState(false);
+  const [infraAutoFixes, setInfraAutoFixes] = useState<Operation[] | undefined>(undefined);
+
+  const [getInfraAutoFixes] = osrdEditoastApi.endpoints.getInfraByInfraIdAutoFixes.useLazyQuery();
+
+  useEffect(() => {
+    getInfraAutoFixes({ infraId: infraId as number })
+      .unwrap()
+      .then((results) => {
+        setInfraAutoFixes(results);
+        setIsLoading(false);
+      });
+  }, []);
+
+  function displayFixes() {
+    if (!infraAutoFixes || isLoading || isSaving) {
+      return (
+        <Loader
+          msg={t(
+            `Editor.infra-errors.corrector-modal.${isLoading ? 'loading-loader' : 'saving-loader'}`
+          )}
+        />
+      );
+    }
+    if (infraAutoFixes?.length === 0) {
+      return <div>{t('Editor.infra-errors.corrector-modal.total-fixes_zero')}</div>;
+    }
+    const creationCount = infraAutoFixes.filter(
+      (operation) => operation.operation_type === OPERATION_TYPE.CREATE
+    ).length;
+    const updateCount = infraAutoFixes.filter(
+      (operation) => operation.operation_type === OPERATION_TYPE.UPDATE
+    ).length;
+    const deletionCount = infraAutoFixes.length - creationCount - updateCount;
```

Comment:
I will let the maintainer decide for that but it might be interesting to store this variables in a useMemo outside of this function to avoid multiple computings (although the component is very small, so not sure if it's worth)
```suggestion
   const [creationCount, updateCount, deletionCount] = useMemo(() => {
    let creation_count = 0;
    let update = 0;
    let deletion = 0;
    if (infraAutoFixes) {
      creation_count = infraAutoFixes.filter(
        (operation) => operation.operation_type === OPERATION_TYPE.CREATE
      ).length;
      update = infraAutoFixes.filter(
        (operation) => operation.operation_type === OPERATION_TYPE.UPDATE
      ).length;
      deletion = infraAutoFixes.length - creation_count - update;
    }
    return [creation_count, update, deletion];
  }, [infraAutoFixes]);
```

Reply:
Since `infraAutoFixes` is recalculated at every render, even with `useMemo`, I believe this would still recompute at every render. So I'd say it's not really useful in this specific case. As you said, we'll let a maintainer shed some light on the matter.

Reply:
I'm afraid that, as said, `useMemo` won't prevent anything from re-render here... and for the sake of complexity of the component, we could let this as this.

Reply:
Actually if you close and open several time the magic wand modal, the infraAutoFixes endpoint isn't called multiple times if the errors are the same. But probably useMemo doesn't make a deep comparison of the dependency array and only use reference which changes every time as you say yes.
Probably the useMemo would have a point if the modal was bigger and had other interactions inside it. That's why I think it might not be needed after all.